pkg/helm/actions/release_history_test.go

```diff
@@ -0,0 +1,84 @@
+package actions
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chartutil"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+)
+
+func TestGetReleaseHistory(t *testing.T) {
+	tests := []struct {
+		name          string
+		releaseName   string
+		release       release.Release
+		err           error
+		maxVersion    int
+		createRelease bool
+	}{
+		{
+			name: "valid release rollback",
+			release: release.Release{
+				Version: 1,
+				Name:    "valid-release",
+				Info: &release.Info{
+					Status: release.StatusDeployed,
+				},
+			},
+			err:           nil,
+			maxVersion:    2,
+			createRelease: true,
+		},
+		{
+			name: "invalid release",
+			release: release.Release{
+				Name: "invalid-release",
+			},
+			maxVersion: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// create fake release
+			store := storage.Init(driver.NewMemory())
+			actionConfig := &action.Configuration{
+				Releases:     store,
+				KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+				Capabilities: chartutil.DefaultCapabilities,
+				Log:          func(format string, v ...interface{}) {},
+			}
+
+			if tt.createRelease {
```

Comment:
Again, if you need to make conditional steps in fixture, then the tests do not have much in common, and I would write two separate test functions.

Reply:
Done